services/slam/collector.go

```diff
@@ -0,0 +1,59 @@
+package slam
+
+import (
+	"context"
+
+	"go.opencensus.io/trace"
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/spatialmath"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+type method int64
+
+const (
+	getPosition      method = iota
+	getPointCloudMap method = iota
+)
+
+func (m method) String() string {
+	switch m {
+	case getPosition:
+		return "GetPosition"
+	case getPointCloudMap:
+		return "GetPointCloudMap"
+	}
+	return "Unknown"
+}
+
+type Position struct {
+	Pose               spatialmath.Pose
+	ComponentReference string
+}
+
+func newGetPositionCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
+	slam, err := assertSLAM(resource)
+	if err != nil {
+		return nil, err
+	}
+
+	cFunc := data.CaptureFunc(func(ctx context.Context, _ map[string]*anypb.Any) (interface{}, error) {
+		_, span := trace.StartSpan(ctx, "slam::data::collector::CaptureFunc::GetPosition")
```

Comment:
Not sure what the naming convention should be for this span, so let me know if this makes sense

Reply:
Do we need a span here? Doesn't GetPosition already get measured?

Reply:
This starts a new child span specifically for data capture

Reply:
don't feel strongly if we have the span for GetPosition here, but I think the SLAM team would def want the span for the GetPointCloudMap data capture.

Reply:
What would it be used for?

Reply:
debugging, since a GetPointCloudMap call can sometimes take 5-10 seconds depending on how large the map was within the slam algorithm, so seeing that performance would be useful, while I feel the performance on GetPosition would be less relevant, as the serialization there is pretty quick

Reply:
sgtm, will make a note to include span for `GetPointCloudMap`, but removed here

Reply:
Doesn't GetPointCloudMap already have a span?

Reply:
It doesn't hurt to include this as a parent span, moreso just wondering what the motivation would be